calc/vxlan_resolver.go

```diff
@@ -263,6 +265,58 @@ func (c *VXLANResolver) OnHostConfigUpdate(update api.Update) (_ bool) {
 			})
 			c.sendVTEPRemove(nodeName)
 		}
+	case "MacVXLANTunnelAddr":
+		nodeName := update.Key.(model.HostConfigKey).Hostname
+		vtepSent := c.vtepSent(nodeName)
+		logCxt := logrus.WithField("node", nodeName).WithField("value", update.Value)
+		logCxt.Debug("MacVXLANTunnelAddr update")
+		if update.Value != nil {
+			// Update for a VXLAN tunnel MAC address.
+			newMAC := update.Value.(string)
+			currMAC := c.nodeNameToVXLANMac[nodeName]
+			logCxt = logCxt.WithFields(logrus.Fields{"newMAC": newMAC, "currMAC": currMAC})
+			if vtepSent {
+				if currMAC == newMAC {
+					// If we've already handled this node, there's nothing to do. Deduplicate.
+					logCxt.Debug("Skipping duplicate tunnel MAC addr update")
+					return
+				}
+
+				// We've already sent a VTEP for this node, and the node's MAC address has changed.
+				// We need to revoke it and any routes before sending any updates.
```

Comment:
As discussed, for the MAC address case, there's no need to revoke the routes since the MAC only affects the VTEP.

Reply:
I think that means this whole block can be simplified a lot vs the IP case.  The MAC isn't required to send the VTEP in the same way that the IP address is required so adding/removing the MAC can never invalidate the routes, it can only update the VTEP (if and only if the VTEP has already been sent).

Reply:
I have updated the code.